mbuild/formats/lammpsdata.py

```diff
@@ -199,9 +203,14 @@ def write_lammpsdata(structure, filename, atom_style='full',
 
 
     # Check impropers
+    dihedrals = []
+    imp_dihedrals  = []
     for dihedral in structure.dihedrals:
```

Comment:
I think you can write this code as. If you can, it is more consistent with the code style for the bonds/angles/impropers.

```python
imp_dihedrals = [[
    dihedral.atom1.idx+1,
    dihedral.atom2.idx+1,
    dihedral.atom3.idx+1,
    dihedral.atom4.idx+1] for dihedral in structure.dihedral if dihedral.improper]

dihedrals = [[
    dihedral.atom1.idx+1,
    dihedral.atom2.idx+1,
    dihedral.atom3.idx+1,
    dihedral.atom4.idx+1] for dihedral in structure.dihedral if not dihedral.improper]
```

Reply:
We also need to triple check the correct atom ordering here. Sometimes there is a central-atom-third convention for AMBER-style impropers.

Reply:
You're right.  I didn't know whether I could include logical conditions in list comprehensions.

Reply:
> We also need to triple check the correct atom ordering here. Sometimes there is a central-atom-third convention for AMBER-style impropers.

There is a central-atom-third convention, and it's already implemented in the write statement later in the code for both styles of impropers.  See lines 534-545.

Reply:
You may want to have a look at what I do [here](https://github.com/mosdef-hub/mbuild/blob/625f956b12e92cdc6bbb9f9227bf01a23911ae93/mbuild/formats/cassandramcf.py#L566) in the MCF writer. You can also look at the openmm issue. Let me know if you have any questions.

Reply:
So I should change it from the style that was already used before my changes?  I did it that way to maintain consistency with how the harmonic impropers were done.  Either way, the atoms are written to the LAMMPS data file in the correct order, corrected from the central-atom-third convention to the central-atom-first convention used by LAMMPS.

Reply:
The `harmonic` vs. `amber` impropers are two very different things, so we need to keep those separate.

Questions:
1. What order does `parmed` use for the harmonic impropers (central atom first or third)
2. What order does `parmed` use for the amber-style impropers (central atom first or third)

LAMMPS uses central-atom-first for harmonic impropers, so thats the order the data file should be written.
LAMMPS uses central-atom-first for cvff impropers, so that is the order the data file should be written. 

My comment refers to the ordering of the remaining atom (j, k, l). In the case of a `harmonic` improper, the order does not matter. However, for a `cvff` improper, the energy will be different depending on the order you use for `j, k, and l`. In order to overcome this ambiguity, AMBER defines a specific atom order, based upon the atom types. This is what I was referring to.

Reply:
Due to the reasons given [here](https://github.com/mosdef-hub/mbuild/pull/926), the improper atoms will remain in the same order as obtained from the parmed structure.